[BUG] Terramate Change Detection panics when module references source outside of root directory

**Describe the bug**
We have a "templates"-folder that we use as a base with "terramate experimental clone". The Terraform code in there is only valid after it's cloned to the target directory. It contains references to locals that don't yet exist, and it contains module source-references that **assume the target directory**, so they can't actually be resolved from the "templates"-folder.

When running `terramate list --changed`, we receive the following panic.

**To Reproduce**

* Create a stack "/stacks/stack-a"
* Create a terraform module call that references the (non-existing) source "../../../foo"
* Run `terramate list --changed`

**Expected behavior**
Don't panic and instead show a better explanation of what went wrong and which source file is the origin of the issue.

**Log Output**

```
panic: abspath "/modules/some-module" does not start with root "/work/"

goroutine 1 [running]:
github.com/terramate-io/terramate/project.PrjAbsPath({0xc0009808a0, 0x5}, {0xc0008c9368, 0x15})
        /home/runner/work/terramate/terramate/project/project.go:132 +0x285
github.com/terramate-io/terramate/stack.(*Manager).tfModuleChanged(0xc00003efc0, {{0xc000500460?, 0xc000810808?}}, {0xc00143ffb0, 0x28}, {0x19af717, 0x5}, 0xc00147d858)
        /home/runner/work/terramate/terramate/stack/manager.go:572 +0x154
github.com/terramate-io/terramate/stack.(*Manager).ListChanged.func1({0xc001016f46, 0x7})
        /home/runner/work/terramate/terramate/stack/manager.go:331 +0x2f4
github.com/terramate-io/terramate/stack.(*Manager).filesApply(0xc00003efc0, {{0xc00131f680?, 0xc00131f680?}}, 0xc00147e8a8)
        /home/runner/work/terramate/terramate/stack/manager.go:546 +0x502
github.com/terramate-io/terramate/stack.(*Manager).ListChanged(0xc00003efc0, {{0x19af717?, 0x1559615?}, 0x0?, 0x0?})
        /home/runner/work/terramate/terramate/stack/manager.go:318 +0x1c8d
github.com/terramate-io/terramate/cmd/terramate/cli.(*cli).listStacks(0xc00063c000, 0x0?, {0x0, 0x0}, {0x0?, 0x0?, 0x0?}, 0x0?)
        /home/runner/work/terramate/terramate/cmd/terramate/cli/cli.go:1453 +0x9e
github.com/terramate-io/terramate/cmd/terramate/cli.(*cli).printStacks(0xc00063c000)
        /home/runner/work/terramate/terramate/cmd/terramate/cli/cli.go:1974 +0x2bd
github.com/terramate-io/terramate/cmd/terramate/cli.(*cli).run(0xc00063c000)
        /home/runner/work/terramate/terramate/cmd/terramate/cli/cli.go:729 +0x83a
github.com/terramate-io/terramate/cmd/terramate/cli.Exec({0x19b2bb8, 0x6}, {0xc00003e3a0, 0x2, 0x2}, {0x20121a0, 0xc000074048}, {0x20121c0, 0xc000074050}, {0x20121c0, ...})
        /home/runner/work/terramate/terramate/cmd/terramate/cli/cli.go:389 +0xfc
main.main()
        /home/runner/work/terramate/terramate/cmd/terramate/main.go:20 +0x89
```

**Environment (please complete the following information):**
 - OS: Linux (WSL)
 - OS Version: Ubuntu 24.04 (WSL)
 - Git Version: 2.43.0
 - Terramate Version: 0.11.4

**Additional context**
We have been able to work around this by placing a ".tmskip" file in our templates-folder, but it would still be nice to get a better error message.

Thanks for reporting @cwe1ss 

I was able to reproduce the issue and I will provide a fix ASAP.

```
[i4k@i4k-desktop 2012 (i4k-test-bug-2)]$ terramate list --changed
panic: abspath "/home/i4k/src/reproduce-bugs/foo" does not start with root "/home/i4k/src/reproduce-bugs/2012/"

goroutine 1 [running]:
github.com/terramate-io/terramate/project.PrjAbsPath({0xc0007c20c0, 0x21}, {0xc000500360, 0x20})
	/home/i4k/src/terramate/project/project.go:132 +0x285
github.com/terramate-io/terramate/stack.(*Manager).tfModuleChanged(0xc00072e4b0, {{0xc000914f20?, 0x10?}}, {0xc000133cb0, 0x30}, {0x19ec581, 0x4}, 0xc000767f98)
	/home/i4k/src/terramate/stack/manager.go:573 +0x14b
github.com/terramate-io/terramate/stack.(*Manager).ListChanged.func1({0xc00073b1b7, 0x7})
	/home/i4k/src/terramate/stack/manager.go:332 +0x2f4
github.com/terramate-io/terramate/stack.(*Manager).filesApply(0xc00072e4b0, {{0xc0009140c0?, 0xc0009140c0?}}, 0xc000768b40)
	/home/i4k/src/terramate/stack/manager.go:547 +0x4f4
github.com/terramate-io/terramate/stack.(*Manager).ListChanged(0xc00072e4b0, {{0x19ec581?, 0x0?}, 0x0?, 0x0?})
	/home/i4k/src/terramate/stack/manager.go:319 +0x1c6d
github.com/terramate-io/terramate/cmd/terramate/cli.(*cli).listStacks(0xc0001d1c00, 0x0?, {0x0, 0x0}, {0x0?, 0x0?, 0x0?}, 0x0?)
	/home/i4k/src/terramate/cmd/terramate/cli/cli.go:1449 +0x9e
github.com/terramate-io/terramate/cmd/terramate/cli.(*cli).printStacks(0xc0001d1c00)
	/home/i4k/src/terramate/cmd/terramate/cli/cli.go:1970 +0x2aa
github.com/terramate-io/terramate/cmd/terramate/cli.(*cli).run(0xc0001d1c00)
	/home/i4k/src/terramate/cmd/terramate/cli/cli.go:729 +0x7f2
github.com/terramate-io/terramate/cmd/terramate/cli.Exec({0x19f792c, 0xa}, {0xc00011a4c0, 0x2, 0x2}, {0x1d98d80, 0xc000136088}, {0x1d98da0, 0xc000136090}, {0x1d98da0, ...})
	/home/i4k/src/terramate/cmd/terramate/cli/cli.go:389 +0xfc
main.main()
	/home/i4k/src/terramate/cmd/terramate/main.go:20 +0x89

```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -29,6 +29,10 @@ Given a version number `MAJOR.MINOR.PATCH`, we increment the:
   - The API Key configuration has precedence over all other authentication methods.
 - Add support for unprefixed Bitbucket URLs in Terraform `module.source`.
 
+### Fixed
+
+- Fix `terramate list --changed` panic when a Terraform module references a path outside the project root.
+
 ## v0.11.4
 
 ### Added

--- a/stack/manager.go
+++ b/stack/manager.go
@@ -10,6 +10,7 @@ import (
 	"path"
 	"path/filepath"
 	"sort"
+	"strings"
 
 	"github.com/rs/zerolog/log"
 	"github.com/terramate-io/terramate/config"
@@ -570,6 +571,14 @@ func (m *Manager) tfModuleChanged(
 	}
 
 	modAbsPath := filepath.Join(basedir, mod.Source)
+	rootdir := m.root.Tree().RootDir()
+	if modAbsPath != rootdir && !strings.HasPrefix(modAbsPath, rootdir+string(filepath.Separator)) {
+		printer.Stderr.WarnWithDetails("skipping module call outside of the root directory", errors.E(
+			"module at %q references path %s (abspath %s) outside of the project root",
+			basedir, mod.Source, modAbsPath,
+		))
+		return false, "", nil
+	}
 	modPath := project.PrjAbsPath(m.root.HostDir(), modAbsPath)
 
 	st, err := os.Stat(modAbsPath)

--- a/stack/manager_test.go
+++ b/stack/manager_test.go
@@ -146,6 +146,14 @@ func TestListChangedStacks(t *testing.T) {
 			},
 		},
 		{
+			name:        "single stack: dependent module changed has source path outside project root",
+			repobuilder: singleStackDependentModuleChangedWithSourceOutsideRepo,
+			want: listTestResult{
+				list:    []string{"/stack"},
+				changed: []string{},
+			},
+		},
+		{
 			name:        "multiple stack: single module changed",
 			repobuilder: multipleStackOneChangedModule,
 			want: listTestResult{
@@ -705,6 +713,49 @@ module "module2" {
 	assert.NoError(t, g.Add(mainFile), "add main.tf")
 	assert.NoError(t, g.Commit("commit main.tf"), "commit main.tf")
 
+	return repo
+}
+
+func singleStackDependentModuleChangedWithSourceOutsideRepo(t *testing.T) repository {
+	repo := singleMergeCommitRepoNoStack(t)
+
+	modules := test.Mkdir(t, repo.Dir, "modules")
+	module1 := test.Mkdir(t, modules, "module1")
+	module2 := test.Mkdir(t, modules, "module2")
+
+	repo.modules = append(repo.modules, module1, module2)
+
+	stack := test.Mkdir(t, repo.Dir, "stack")
+	root, err := config.LoadRoot(repo.Dir)
+	assert.NoError(t, err)
+	createStack(t, root, stack)
+	g := test.NewGitWrapper(t, repo.Dir, []string{})
+
+	test.WriteFile(t, stack, "main.tf", `
+module "module1" {
+	source = "../modules/module1"
+}
+`)
+	test.WriteFile(t, module1, "main.tf", `
+module "module2" {
+	source = "../../../../../../../module2"
+}
+`)
+
+	test.WriteFile(t, module2, "main.tf", `# nothing here`)
+
+	assert.NoError(t, g.Add(repo.Dir), "add files")
+	assert.NoError(t, g.Commit("files"), "commit files")
+	assert.NoError(t, g.Push("origin", "main"))
+
+	assert.NoError(t, g.Checkout("change-module", true))
+
+	test.WriteFile(t, module2, "main.tf", `
+# any change
+`)
+
+	assert.NoError(t, g.Add(repo.Dir), "add main.tf")
+	assert.NoError(t, g.Commit("commit module1/main.tf"), "commit main.tf")
 	return repo
 }
 

